bolt04/blinded-onion-message-onion-test.json

```diff
@@ -0,0 +1,143 @@
+{
+  "comment": "Test vector creating an onionmessage, including joining an existing one",
+  "generate": {
+    "comment": "This sections contains test data for Dave's blinded path Bob->Dave; sender has to prepend a hop to Alice to reach Bob",
+    "session_key": "0303030303030303030303030303030303030303030303030303030303030303",
+    "hops": [
+      {
+        "alias": "Alice",
+        "comment": "Alice->Bob: note next_blinding_override to match that give by Dave for Bob",
+        "blinding_secret": "6363636363636363636363636363636363636363636363636363636363636363",
+        "tlvs": {
+          "next_node_id": "0324653eac434488002cc06bbfb7f10fe18991e35f9fe4302dbea6d2353dc0ab1c",
+          "next_blinding_override": "031b84c5567b126440995d3ed5aaba0565d71e1834604819ff9c17f5e9d5dd078f",
+          "blinding_override_secret": "0101010101010101010101010101010101010101010101010101010101010101"
+        },
+        "encrypted_data_tlv": "04210324653eac434488002cc06bbfb7f10fe18991e35f9fe4302dbea6d2353dc0ab1c0821031b84c5567b126440995d3ed5aaba0565d71e1834604819ff9c17f5e9d5dd078f",
+        "ss": "c04d2a4c518241cb49f2800eea92554cb543f268b4c73f85693541e86d649205",
+        "HMAC256('blinded_node_id', ss)": "bc5388417c8db33af18ab7ba43f6a5641861f7b0ecb380e501a739af446a7bf4",
+        "blinded_node_id": "02d1c3d73f8cac67e7c5b6ec517282d5ba0a52b06a29ec92ff01e12decf76003c1",
+        "E": "031195a8046dcbb8e17034bca630065e7a0982e4e36f6f7e5a8d4554e4846fcd99",
+        "H(E || ss)": "83377bd6096f82df3a46afec20d68f3f506168f2007f6e86c2dc267417de9e34",
+        "next_e": "bf3e8999518c0bb6e876abb0ae01d44b9ba211720048099a2ba5a83afd730cad01",
+        "rho": "6926df9d4522b26ad4330a51e3481208e4816edd9ae4feaf311ea0342eb90c44",
+        "encrypted_recipient_data": "49531cf38d3280b7f4af6d6461a2b32e3df50acfd35176fc61422a1096eed4dfc3806f29bf74320f712a61c766e7f7caac0c42f86040125fbaeec0c7613202b206dbdd31fda56394367b66a711bfd7d5bedbe20bed1b"
+      },
+      {
+        "alias": "Bob",
+        "comment": "Bob->Carol",
+        "blinding_secret": "0101010101010101010101010101010101010101010101010101010101010101",
+        "tlvs": {
+          "padding": "0000000000000000000000000000000000000000000000000000",
```

Comment:
Is there a reason to have padding in every hop of Dave's blinded path? It seems like the largest one wouldn't need padding, only the smaller ones, to make them all equal-length

Reply:
Yes, that would have been clever.  Since I'm rebasing anyway, let me take a look...

Reply:
Ah, we did it so that Alice wouldn't need to pad, but Dave wouldn't know that.  Fixing...